test/parallel/test-http-content-length-mismatch.js

```diff
@@ -78,3 +78,23 @@ function shouldThrowOnFewerBytes() {
 shouldThrowOnMoreBytes();
 shouldNotThrow();
 shouldThrowOnFewerBytes();
+
+
+{
+  const server = http.createServer(common.mustCall((req, res) => {
+    res.strictContentLength = true;
+    // Pass content-length as string
```

Comment:
I would move the whole test to a separate file. This more about value type than length mismatch.